fem/pfespace.cpp

```diff
@@ -2598,35 +2599,52 @@ ParFiniteElementSpace::RebalanceMatrix(int old_ndofs,
          }
       }
    }
-   HYPRE_Int* j_offd = make_j_array(i_offd, vsize);
+   HYPRE_BigInt* j_offd = make_j_array(i_offd, vsize);
+
+#ifndef HYPRE_MIXEDINT
+   HYPRE_Int *i_offd_hi = i_offd;
+#else
+   // Copy of i_offd array as array of HYPRE_Int
+   HYPRE_Int *i_offd_hi = Memory<HYPRE_Int>(vsize + 1);
+   std::copy(i_offd, i_offd + vsize + 1, i_offd_hi);
+   Memory<HYPRE_BigInt>(i_offd, vsize + 1, true).Delete();
+#endif
 
    // create the offd column map
-   int offd_cols = i_offd[vsize];
-   Array<Pair<HYPRE_Int, int> > cmap_offd(offd_cols);
+   int offd_cols = i_offd_hi[vsize];
+   Array<Pair<HYPRE_BigInt, int> > cmap_offd(offd_cols);
    for (int i = 0; i < offd_cols; i++)
    {
       cmap_offd[i].one = j_offd[i];
       cmap_offd[i].two = i;
    }
-   SortPairs<HYPRE_Int, int>(cmap_offd, offd_cols);
 
-   HYPRE_Int* cmap = Memory<HYPRE_Int>(offd_cols);
+#ifndef HYPRE_MIXEDINT
+   HYPRE_Int *j_offd_hi = j_offd;
+#else
+   HYPRE_Int *j_offd_hi = Memory<HYPRE_Int>(offd_cols);
+   Memory<HYPRE_BigInt>(j_offd, offd_cols, true).Delete();
```

Comment:
Is `j_offd` created and then deleted? Can this be simplified?

Reply:
This is a trick to delete a host pointer that was allocated via the memory manager (using the currently set host allocator). The respective allocation is done with something like `HYPRE_BigInt *j_offd = Memory<HYPRE_BigInt>(size);` -- you can check above that `j_offd` was allocated that way.